### PR TITLE
Updating EP docs with Onnxruntime API calls

### PR DESCRIPTION
**Pull request by oliviajain**
**Description**: Updating EP Docs to use Onnxruntime API

**Motivation and Context**
- No longer using internal onnx but referencing public API 
- Allows users to leverage EPs correctly

### CODE REVIEW

**Review by natke — DISMISSED** [review]

File: docs/execution_providers/RKNPU-ExecutionProvider.md [hunk 1]
```diff
@@ -15,17 +15,10 @@ For build instructions, please see the [BUILD page](../../BUILD.md#RKNPU).
...
-                                  LoggingManager::InstanceType::Default,
-                                  &log_id)
-Environment::Create(std::move(logging_manager), env)
-InferenceSession session_object{so,env};
-session_object.RegisterExecutionProvider(std::make_unique<::onnxruntime::RknpuExecutionProvider>());
-status = session_object.Load(model_file_name);
+Ort::Env env = Ort::Env{ORT_LOGGING_LEVEL_ERROR, "Default"};
+Ort::SessionOptions sf;
+bool enable_cpu_mem_arena = true;
+Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_RKNPU(sf, enable_cpu_mem_arena));
```
Comment (oliviajain) [inline docs/execution_providers/RKNPU-ExecutionProvider.md:21]: Need to build/test this tutorial out, wasn't in onnx test runner.

File: docs/execution_providers/ACL-ExecutionProvider.md [hunk 2]
```diff
@@ -7,19 +7,11 @@ For build instructions, please see the [BUILD page](../../BUILD.md#ARM-Compute-L
...
-(std::unique_ptr<ISink>{new CLogSink{}},
-                                  static_cast<Severity>(lm_info.default_warning_level),
-                                  false,
-                                  LoggingManager::InstanceType::Default,
-                                  &log_id)
-Environment::Create(std::move(logging_manager), env)
-InferenceSession session_object{so, env};
-session_object.RegisterExecutionProvider(std::make_unique<::onnxruntime::ACLExecutionProvider>());
-status = session_object.Load(model_file_name);
+Ort::Env env = Ort::Env{ORT_LOGGING_LEVEL_ERROR, "Default"};
```
Comment (jywu-msft) [inline docs/execution_providers/ACL-ExecutionProvider.md:11]: the previous internal C++ api examples create a session.
shouldn't we do the same for the corrected examples?
create a session with the session options that you initialized with the various EP options?

**Comment by pranavsharma** [comment]
I suggest we add a file here (\docs\execution_providers) for the CUDA EP as well since people are [finding it difficult](\docs\execution_providers) to find documentation on how to use it.

**Comment by jywu-msft** [comment]
> I suggest we add a file here (docs\execution_providers) for the CUDA EP as well since people are [finding it difficult](https://github.com/microsoft/onnxruntime/issues/5512#issuecomment-710060065) to find documentation on how to use it.

I think that's outside the scope of this change and can be done separately.

**Review by jywu-msft — APPROVED** [review]
Thanks!

**Pull request merged**